engine/src/test/java/io/zeebe/engine/processing/streamprocessor/StreamProcessorReprocessingTest.java

```diff
@@ -87,6 +92,85 @@ public void shouldCallRecordProcessorLifecycle() {
     inOrder.verifyNoMoreInteractions();
   }
 
+  @Test
+  public void shouldNotSwitchToProcessingAfterReprocessingWhenPaused() throws Exception {
+    // given - bunch of events to reprocess
+    IntStream.range(0, 5000)
+        .forEach(i -> streamProcessorRule.writeWorkflowInstanceEvent(ELEMENT_ACTIVATING, i));
+    final long sourceEvent = streamProcessorRule.writeWorkflowInstanceEvent(ELEMENT_ACTIVATING, 1);
+    streamProcessorRule.writeWorkflowInstanceEventWithSource(
+        WorkflowInstanceIntent.ELEMENT_ACTIVATED, 1, sourceEvent);
+
+    Awaitility.await()
+        .until(
+            () ->
+                streamProcessorRule
+                    .events()
+                    .onlyWorkflowInstanceRecords()
+                    .withIntent(ELEMENT_ACTIVATED),
+            StreamWrapper::exists);
+
+    // when
+    final var countDownLatch = new CountDownLatch(1);
+    final var streamProcessor =
+        streamProcessorRule.startTypedStreamProcessor(
+            (processors, context) ->
+                processors.withListener(
+                    new StreamProcessorLifecycleAware() {
+                      @Override
+                      public void onRecovered(final ReadonlyProcessingContext context) {
+                        countDownLatch.countDown();
+                      }
+                    }));
+    streamProcessor.pauseProcessing();
```

Comment:
Please add some visual separation. I oversaw the statement :sweat_smile:

Reply:
valid point :D